testing/deployer/topology/topology.go

```diff
@@ -753,9 +780,11 @@ type Service struct {
 	Command []string `json:",omitempty"` // optional
 	Env     []string `json:",omitempty"` // optional
 
-	DisableServiceMesh bool `json:",omitempty"`
-	IsMeshGateway      bool `json:",omitempty"`
-	Upstreams          []*Upstream
+	EnableTransparentProxy bool        `json:",omitempty"`
+	DisableServiceMesh     bool        `json:",omitempty"`
+	IsMeshGateway          bool        `json:",omitempty"`
+	Upstreams              []*Upstream `json:",omitempty"`
+	ImpliedUpstreams       []*Upstream `json:",omitempty"`
```

Comment:
So that the `Relationship` trick can continue to be used to automatically make some basic assertions about upstreams, I added `ImpliedUpstreams` to be where you can tell `deployer` about these implicitly reachable destinations, but no machinery will use those to do anything. They will however get all of the denormalized metadata injected in them to facilitate actually talking to them during assertion.